data-plane/src/server/server.rs

```diff
@@ -0,0 +1,233 @@
+        UnboundedSender<LogHandlerMessage>,
+        UnboundedReceiver<LogHandlerMessage>,
+    ) = unbounded_channel();
+
+    let feature_context = Arc::new(context);
+    if feature_context.trx_logging_enabled {
+        let tx_for_handler = tx.clone();
+        tokio::spawn(async move {
+            start_log_handler(tx_for_handler, rx).await;
+        });
+    }
+
+    log::info!("TLS Server Created - Listening for new connections.");
+    let cage_context = match CageContext::get() {
+        Ok(context) => Arc::new(context),
+        Err(e) => {
+            log::error!("Failed to read cage context in data plane server - {e}");
+            return;
+        }
+    };
+    let service_builder = tower::ServiceBuilder::new();
+
+    // Only apply attestation layer in enclave mode
+    #[cfg(feature = "enclave")]
+    let service_builder = service_builder.layer(AttestLayer);
+
+    // layers are invoked in the order that they're registered to the service
+    let service = service_builder
+        .layer(ContextLogLayer::new(
+            cage_context.clone(),
+            feature_context.clone(),
+            tx.clone(),
+        ))
+        .option_layer(
+            feature_context
+                .api_key_auth
+                .then(|| AuthLayer::new(e3_client.clone(), cage_context.clone())),
+        )
+        .layer(DecryptLayer::new(e3_client.clone()))
+        .service(ForwardService);
+    loop {
+        let mut stream = match server.accept().await {
+            Ok(stream) => stream,
+            Err(tls_err) => {
+                log::error!(
+                    "An error occurred while accepting the incoming connection — {tls_err}"
+                );
+                continue;
+            }
+        };
+
+        let remote_ip = stream.get_remote_addr().clone();
+        let tx_for_connection = tx.clone();
+        let mut data_plane_service = service.clone();
+        let cage_context_clone = cage_context.clone();
+        let feature_context_clone = feature_context.clone();
+        let e3_client_clone = e3_client.clone();
+        tokio::spawn(async move {
+            loop {
+                match try_parse_http_request_from_stream(&mut stream, port).await {
+                    Ok(Incoming::HttpRequest(request)) if parse::is_websocket_request(&request) => {
+                        return handle_websocket_request(
+                            &mut stream,
+                            request,
+                            &tx_for_connection,
+                            remote_ip,
+                            cage_context_clone.clone(),
+                            feature_context_clone.clone(),
+                            e3_client_clone.clone(),
+                            port,
+                        )
+                        .await;
+                    }
+                    Ok(Incoming::HttpRequest(request)) => {
+                        let response = data_plane_service.call(request).await.unwrap();
```

Comment:
Would be good to handle the error here and shutdown the connection

Reply:
Good spot, yeah can convert that error into a proper response!